nmostesting/suites/IS1102Test.py

```diff
@@ -13,16 +13,1402 @@
+            ):
+                return "YCbCr-4:2:0"
+
+        return "invalid array of video components"
+
+    def get_another_grain_rate(self, grain_rate):
+        numerator = grain_rate["numerator"]
+        denominator = grain_rate["denominator"]
+        if (numerator == 30 or numerator == 25) and denominator == 1:
+            return {"numerator": numerator * 2, "denominator": 1}
+        if (numerator == 60 or numerator == 50) and denominator == 1:
+            return {"numerator": numerator / 2, "denominator": 1}
+        if numerator == 24 and denominator == 1:
+            return {"numerator": 30, "denominator": 1}
+        if (numerator == 30000 or numerator == 25000) and denominator == 1001:
+            return {"numerator": numerator * 2, "denominator": 1001}
+        if (numerator == 60000 or numerator == 50000) and denominator == 1001:
+            return {"numerator": numerator / 2, "denominator": 1001}
+        return "grain_rate not valid"
+
+    def get_another_sample_rate(self, sample_rate):
+        numerator = sample_rate["numerator"]
+        if numerator == 48000:
+            return {"numerator": 44100}
+        if numerator == 44100:
+            return {"numerator": 48000}
+        if numerator == 96000:
+            return {"numerator": 4800}
+        if numerator == 88200:
+            return {"numerator": 44100}
+        return "sample_rate not valid"
+
+    def test_02_00(self, test):
+        "Reset active constraints of all senders"
+        _, response = TestHelper.do_request("GET", self.compat_url + "senders/")
+        if response.status_code == 200:
+            self.senders = response.json()
+            for sender in self.senders:
+                _, response = TestHelper.do_request(
+                    "DELETE",
+                    self.compat_url + "senders/" + sender + "constraints/active/",
+                )
+                if response.status_code != 200:
+                    return test.FAIL("senders constraints cannot be deleted")
+            return test.PASS()
+
+        return test.FAIL(response.json())
+
+    def test_02_01(self, test):
+        "Verify that the device supports the concept of IS-11 Sender"
+        _, response = TestHelper.do_request("GET", self.compat_url + "senders/")
+        if response.status_code != 200:
+            return test.FAIL(response.json)
+        self.senders = response.json()
+        if len(self.senders) == 0:
+            return test.UNCLEAR("there is no IS-11 senders")
+        return test.PASS()
+
+    def test_02_01_01(self, test):
+        "Verify that the device supports the concept of IS-11 Sender"
+        if len(self.senders) != 0:
+            for sender_id in self.senders:
+                _, response = TestHelper.do_request(
+                    "GET", self.node_url + "senders/" + sender_id
+                )
+                if response.status_code != 200:
+                    return test.FAIL()
```

Comment:
"return test.FAIL()" is not allowed, it must return a detail parameter. "TypeError: FAIL() missing 1 required positional argument: 'detail'"